### DISCUSSION TITLE
OOM occurred while creating an index with faiss module

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15697] [Q&A and General discussion]
hi, I am trying to create an index (IVFPQ m = 8, nlist = 42342), all data size = 448221860 * 32dim

OOM occurred when memory allocating on GPU, error info shown below:
`Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 173432832 bytes (error 2 out of memory)
CreateIndex function call status: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 173432832 bytes (error 2 out of memory)`

Something must be wrong, pls help me.

**Comment by yhmo** [comment]
What is the value of **index_file_size**? (index_file_size is a parameter of create_collection())

Your GPU has 32GB of memory, definitely is enough to build index for a segment. The possible reason: is any other application occupy the GPU memory?

Observe the GPU state:
`nvidia-smi -lms 500`

**Reply by xiaojuan8** [reply]
Hi, index_file_size = 1024MB.

**Reply by yhmo** [reply]
I guess the GPU memory is occupied by other applications. This error is thrown by Faiss, it could not apply GPU memory by cudaMalloc().

**Reply by xiaojuan8** [reply]
👌got it. I will try my indexing again.
